lib/ansible/modules/cloud/kubevirt/kubevirt_cdi_upload.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+
+module: kubevirt_cdi_upload
+
+short_description: Upload local VM images to CDI Upload Proxy.
+
+version_added: "2.8"
+
+author: KubeVirt Team (@kubevirt)
+
+
+description:
+  - Use Openshift Python SDK to create UploadTokenRequest objects.
+  - Transfer contents of local files to the CDI Upload Proxy.
+
+options:
+  pvc_name:
+    description:
+      - Use to specify the name of the target PersistentVolumeClaim.
+    required: true
+  pvc_namespace:
+    description:
+      - Use to specify the namespace of the target PersistentVolumeClaim.
+    required: true
+  upload_host:
+    description:
+      - URL containing the host and port on which the CDI Upload Proxy is available.
+      - "More info: U(https://github.com/kubevirt/containerized-data-importer/blob/master/doc/upload.md#expose-cdi-uploadproxy-service)"
+  upload_host_verify_ssl:
```

Comment:
Is this different than the equivalent option specified in k8s_auth_options?  Does this module connect to two API endpoints, and you could have two different cert-validation settings?  Can we re-use the one defined in k8s_auth_options here?

Reply:
Yes, this is completely different. Stuff in k8s_auth_options controls connectivity to the kubernetes cluster's api server endpoint. This module does that, but then also connects to a service running _on_ the cluster itself (called the Containerized Data Importer Upload Proxy), which is why it has a completely separate set of `upload_host_*` parameters that control that.

It's probable that before The Freeze I'll be attempting to switch the default on `upload_host_verify_ssl` to _autodetect_ (aka "look for a Secret with the certs in a standard place") for ease of use.

Reply:
Thanks for that clarification.  Upon reviewing the code, it wasn't obvious to me that this module is actively using both connection paths (to the k8s API and also to the Upload Proxy.)  I've taken your word for it and updated my related PR.